src/NosnocSolver.m

```diff
@@ -515,10 +1012,28 @@ function  print_solver_stats(obj, results, stats)
                 W = [W,results.w];
             end
 
+            if solver_options.calculate_stationarity_type
+                if last_iter_failed || ~obj.complementarity_tol_met(stats) || timeout
+                    stat_type = "?";
+                    disp("Not checking stationarity due to failure of homotopy to converge");
+                else
+                    [polished_w, res_out, stat_type, n_biactive] = obj.calculate_stationarity(results.nlp_results(end), false);
+                end
+                if stat_type ~= "?"
```

Comment:
It can but as `calculate_stationarity` may also return `stat_type = "?"` we still need the if statement. We could move it inside the else but I think it is a little more readable as the above block generating the type and then we choose what to do with it.